src/mpeg4/mpeg4Boxes.ts

```diff
@@ -0,0 +1,1952 @@
+        if (this.children === null || this.children === undefined) {
+            return undefined;
+        }
+
+        for (const box of this.children) {
+            if (ByteVector.equals(box.boxType, type)) {
+                return box;
+            }
+        }
+
+        for (const box of this.children) {
+            const childBox: Mpeg4Box = box.getChildRecursively(type);
+
+            if (childBox !== null && childBox !== undefined) {
+                return childBox;
+            }
+        }
+
+        return undefined;
+    }
+
+    /**
+     * Removes all children with a specified box type from the current instance.
+     * @param type A @see ByteVector object containing the box type to remove.
+     */
+    public removeChildByType(type: ByteVector): void {
+        if (this.children === null || this.children === undefined) {
+            return;
+        }
+
+        for (const box of this.children) {
+            if (ByteVector.equals(box.boxType, type)) {
+                const index = this.children.indexOf(box);
+
+                if (index > -1) {
+                    this.children.splice(index, 1);
+                }
+            }
+        }
+    }
+
+    /**
+     * Removes a specified box from the current instance.
+     * @param box A @see Mpeg4Box object to remove from the current instance.
+     */
+    public removeChildByBox(box: Mpeg4Box): void {
+        if (this.children === null || this.children === undefined) {
+            return;
+        }
+
+        const index = this.children.indexOf(box);
+
+        if (index > -1) {
+            this.children.splice(index, 1);
+        }
+    }
+
+    /**
+     * Adds a specified box to the current instance.
+     * @param box A @see Mpeg4Box object to add to the current instance.
+     */
+    public addChild(box: Mpeg4Box): void {
+        if (this.children === null || this.children === undefined) {
+            return;
+        }
+
+        this.children.push(box);
+    }
+
+    /**
+     * Removes all children from the current instance.
+     */
+    public clearChildren(): void {
+        if (this.children === null || this.children === undefined) {
+            return;
+        }
+
+        this.children = [];
+    }
+
+    /**
+     * Loads the children of the current instance from a specified file using the internal data position and size.
+     * @param file The File from which the current instance was read and from which to read the children.
+     * @returns A @see Mpeg4Box[] object enumerating the boxes read from the file.
+     */
+    public loadChildren(file: File): Mpeg4Box[] {
+        Guards.notNullOrUndefined(file, "file");
+
+        const children: Mpeg4Box[] = [];
+
+        let position: number = this.dataPosition;
+        const end: number = position + this.dataSize;
+
+        this._header.box = this;
+
+        while (position < end) {
+            const child: Mpeg4Box = Mpeg4BoxFactory.createBoxFromFilePositionParentHandlerAndIndex(

```

Comment:
Agreed. I see this as a code smell. The whole box inheritance and creation is a mess. I tried, but I wasn't able to come to a solution where a dependency to the factory is not required. I believe this would require a rather extensive redesign of the whole box inheritance and creation architecture. Is this something we could consider as a TODO?

Reply:
Definitely, we can that as TODO 👍